python/ngen_cal/src/ngen/cal/_hookspec.py

```diff
@@ -42,3 +44,21 @@ def ngen_cal_finish(exception: Exception | None) -> None:
     raised during the calibration loop.
     `exception` will be non-none if an exception was raised during calibration.
     """
+
+class ModelHooks():
+    @hookspec(firstresult=True)
+    def ngen_cal_model_output(id: str | None) -> Series:
```

Comment:
Wondering if the weird behavior you were seeing with hooks returning more than a single value was because you are missing `self` here?

```suggestion
    def ngen_cal_model_output(self, id: str | None) -> Series:
```

Reply:
Fixed in 082e2712272f339a94c5d01fa8a99f33a553b1ea, behavior didn't change...